### PR TITLE
Handle empty OpenAI model listings as cache miss

### PR DESCRIPTION
**Pull request by FrancescoMonti-source**
## Summary
- Avoid returning placeholder rows when OpenAI's model listing is empty
- Add regression test ensuring empty listings yield no rows

## Testing
- `R -q -e "devtools::test()"` *(fails: command not found: R)*
- `apt-get update` *(fails: The repository 'http://security.ubuntu.com/ubuntu noble-security InRelease' is not signed)*

------
https://chatgpt.com/codex/tasks/task_e_68b5aeb601fc8321aa283b55e451342c

### CODE REVIEW

File: R/models_cache.R [hunk 1]
```diff
@@ -388,11 +388,15 @@ list_models <- function(provider = NULL,
...
-        # Cache ONLY if ok and non-empty
-        if (identical(live$status, "ok") && nrow(live$df) > 0) {
-          .cache_put("openai", bu, live$df)
+        if (identical(live$status, "ok") && nrow(live$df) == 0) {
+          live$status <- "empty_cache"
+          Sys.sleep(0.25)
+        } else {
+          if (identical(live$status, "ok") && nrow(live$df) > 0) {
+            .cache_put("openai", bu, live$df)
+          }
```
[the remark this answers is not in the document]
Comment (FrancescoMonti-source) [reply R/models_cache.R:397]: @codex fix comments
